File: aoc_2018/test/day01_test.py
import aoc_2018.test.test_utils as test_utils
import aoc_2018.day01 as day01
import pytest

def test_1():
    assert day01.part1(['+1', '-2', '+3', '+1']) == 3
    assert day01.part1(['+1', '+1', '+1']) == 3
    assert day01.part1(['+1', '+1', '-2']) == 0
    assert day01.part1(['-1', '-2', '-3']) == -6
    my_input = get_my_input()
    assert day01.part1(my_input) == 400

def test_2():
    assert day01.part2(['+1', '-2', '+3', '+1']) == 2
    assert day01.part2(['+1', '-1']) == 0
    assert day01.part2(['+3', '+3', '+4', '-2', '-4']) == 10
    assert day01.part2(['-6', '+3', '+8', '+5', '-6']) == 5
    assert day01.part2(['+7', '+7', '-2', '-7', '-4']) == 14
    my_input = get_my_input()
    assert day01.part2(my_input) == 232

def get_my_input():
    path = test_utils.get_path('day01_input.txt')
    with open(path) as file:
        return [line for line in file]

File: aoc_2018/kt18/src/test/kotlin/com/github/dstine/aoc/kt18/Day01Test.kt
package com.github.dstine.aoc.kt18

import java.io.File
import org.junit.Assert.assertEquals
import org.junit.Test

class Day01Test {

    @Test
    fun test1() {
        assertEquals(part1(listOf("+1", "-2", "+3", "+1")), 3)
        assertEquals(part1(listOf("+1", "+1", "+1")), 3)
        assertEquals(part1(listOf("+1", "+1", "-2")), 0)
        assertEquals(part1(listOf("-1", "-2", "-3")), -6)
        val myInput = getMyInput()
        assertEquals(part1(myInput), 400)
    }
    
    @Test
    fun test2() {
        assertEquals(part2(listOf("+1", "-2", "+3", "+1")),  2)
        assertEquals(part2(listOf("+1", "-1")), 0)
        assertEquals(part2(listOf("+3", "+3", "+4", "-2", "-4")), 10)
        assertEquals(part2(listOf("-6", "+3", "+8", "+5", "-6")), 5)
        assertEquals(part2(listOf("+7", "+7", "-2", "-7", "-4")), 14)
        val myInput = getMyInput()
        assertEquals(part2(myInput), 232)

    }

    private fun getMyInput(): List<String> {
        val path = "../data/day01_input.txt"
        return File(path).useLines { it.toList() }
    }
}
